### DISCUSSION TITLE
The Testing Question

### DISCUSSION THREAD

**Discussion by Mateusz-Dobrzynski** [debatecore/.github#8] [General]
We don't use any testing frameworks at the moment. This is a problem because, as the code base grows and becomes more interconnected, the risk of bugs creation grows. We want functional software (of course), especially since its use case is competitive tournaments. High test coverage is yet another way of gaining trust.

I don't know how to test a UI-heavy project like _[Debate Tools](https://github.com/debatecore/debate-tools)_. One idea coming to my mind is using _[Beautiful Soup](https://www.crummy.com/software/BeautifulSoup/)_, a tool for web scraping. It is, however, a Python library. We could use an equivalent suitable for our tech stack, assuming there is one.

There are, of course, unit tests and we could use them for testing functions called under the hood. That will be crucial for _[tau](https://github.com/debatecore/tau)_, so it's important to start with good practices in mind. But for that, we need to decide how to test. Hence, I'm starting the discussion.

@jakubmanczak, @BoberITman

**Comment by jakubmanczak** [comment]
i'd assume we'd use https://playwright.dev/ for e2e & some react unit testing library

**Comment by Mateusz-Dobrzynski** [comment]
Created an issue for Playwright setup ([#79](https://github.com/debatecore/debate-tools/issues/79)).
For unit tests, I suggest [Jest](https://jestjs.io/).
